src/cwe_checker_lib/src/intermediate_representation/program.rs

```diff
@@ -37,16 +37,87 @@ impl Program {
+
     use super::*;
 
     impl Program {
-        pub fn mock_empty() -> Program {
+        /// Returns Program with malloc, free and other_function
+        pub fn mock_x64() -> Program {
+            let malloc = ExternSymbol {
+                tid: Tid::new("malloc"),
+                addresses: vec![],
+                name: "malloc".to_string(),
+                calling_convention: Some("__stdcall".to_string()),
+                parameters: vec![Arg::from_var(
+                    CallingConvention::mock_x64().integer_parameter_register[0].clone(),
+                    Some(Datatype::Integer),
+                )],
+                return_values: vec![Arg::from_var(
+                    CallingConvention::mock_x64().integer_return_register[0].clone(),
+                    Some(Datatype::Pointer),
+                )],
+                no_return: false,
+                has_var_args: false,
+            };
+            let free = ExternSymbol {
+                tid: Tid::new("free"),
+                addresses: vec![],
+                name: "free".to_string(),
+                calling_convention: Some("__stdcall".to_string()),
+                parameters: vec![Arg::from_var(
+                    CallingConvention::mock_x64().integer_parameter_register[0].clone(),
+                    Some(Datatype::Pointer),
+                )],
+                return_values: vec![],
+                no_return: false,
+                has_var_args: false,
+            };
+            let other_function = ExternSymbol {
+                tid: Tid::new("other_function"),
+                addresses: vec![],
+                name: "other_function".to_string(),
+                calling_convention: None,
+                parameters: vec![],
+                return_values: vec![],
+                no_return: false,
+                has_var_args: false,
+            };
+
             Program {
                 subs: BTreeMap::new(),
-                extern_symbols: BTreeMap::new(),
+                extern_symbols: BTreeMap::from([
+                    (malloc.tid.clone(), malloc),
+                    (free.tid.clone(), free),
+                    (other_function.tid.clone(), other_function),
+                ]),
                 entry_points: BTreeSet::new(),
-                address_base_offset: 0,
+                address_base_offset: 0x1000u64,
+            }
+        }
+        /// Returns Program with malloc, free and other_function
+        pub fn mock_arm32() -> Program {
+            // Creates arm32 program by altering x64 program
+            let mut prog = Program::mock_x64();
+            for symbol in prog.extern_symbols.values_mut() {
+                if symbol.name == "malloc".to_string() {
+                    symbol.parameters = vec![Arg::from_var(
+                        CallingConvention::mock_arm32().integer_parameter_register[0].clone(),
+                        Some(Datatype::Integer),
+                    )];
```

Comment:
While your code works as intended, its structure could backfire in the future if we change something in the `mock_x64` function without realizing that it also affects the `mock_arm32` function. One possible better structure would be using a helper function for creating the extern symbols which gets the calling convention as a parameter.